### PR TITLE
Add/Fix documentation of Public API page

### PR DESCRIPTION
**Pull request by dimitraka**

### CODE REVIEW

File: libs/core/async_base/include/hpx/async_base/dataflow.hpp [hunk 1]
```diff
@@ -4,8 +4,29 @@
...
+    /// \brief The function template \a dataflow runs the function f
+    ///        asynchronously (potentially in a separate thread which might be
+    ///        a part of a thread pool) and returns a \c hpx::future that will
+    ///        eventually hold the result of that function call. Its behavior
+    ///        is similar to \c hpx::async with the exception that if one of
+    ///        the arguments is a \a future, then \c hpx::dataflow will wait
+    ///        for the \a future to be ready to launch the thread. Hence, the
+    ///        operation is delayed until all the arguments are ready.
+    template <typename F, typename... Ts>
+    auto dataflow(F&& f, Ts&&... ts);
```
Comment (dimitraka) [inline libs/core/async_base/include/hpx/async_base/dataflow.hpp:24]: Return type of `dataflow` should be `auto` or `lcos::detail::dataflow_dispatch<typename std::decay<F>::type>`?

File: libs/core/datastructures/include/hpx/datastructures/tuple.hpp [hunk 2]
```diff
@@ -10,6 +10,27 @@
...
+    /// \brief Constructs a tuple that is a concatenation of all tuples in \c tuples.
+    ///        The behavior is undefined if any type in \c std::decay_t<Tuples>...
+    ///        is not a specialization of \c hpx::tuple. However, an implementation
+    ///        may choose to support types (such as \c std::array and \c std::pair)
+    ///        that follow the tuple-like protocol.
+    ///
+    /// \param tuples - zero or more tuples to concatenate
+    ///
+    /// \returns hpx::tuple object composed of all elements of all argument tuples
+    ///          constructed from \c hpx::get<Is>(HPX_FORWARD(UTuple,t) for each
```
Comment (dimitraka) [inline libs/core/datastructures/include/hpx/datastructures/tuple.hpp:26]: Is the phrase "constructed by `hpx::get<Is>(HPX_FORWARD(UTuple,t)`" correct?

File: libs/core/async_base/include/hpx/async_base/dataflow.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/core/async_base/include/hpx/async_base/dataflow.hpp:24]: Let's leave it as `auto` as there are several overloads, all of them might return different things.

File: libs/core/datastructures/include/hpx/datastructures/tuple.hpp (hunk 2 above)
Comment (hkaiser) [reply libs/core/datastructures/include/hpx/datastructures/tuple.hpp:79]: Those are equivalent. `HPX_FORWARD(T, t)` is essentially the same as `std::forward<T>(t)`.

File: libs/core/async_base/include/hpx/async_base/dataflow.hpp (hunk 1 above)
Comment (dimitraka) [reply libs/core/async_base/include/hpx/async_base/dataflow.hpp:24]: Sounds good!

**Comment by hkaiser** [comment]
@dimitraka could you please address the clang-format/inspect complaints?

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**